### ISSUE TITLE
permissions-dispatcher:ktx:1.0.5 has bug

### ISSUE THREAD

**Issue by LZ9** [permissions-dispatcher/PermissionsDispatcher#729]
in PermissionRequestViewModel.kt this function : 
```
    inline fun observe(
        owner: LifecycleOwner,
        key: String,
        crossinline requiresPermission: Fun,
        noinline onPermissionDenied: Fun?,
        noinline onNeverAskAgain: Fun?
    ) {
        permissionRequestResult.observe(owner, Observer {
            when (it[key]?.getContentIfNotHandled()) {
                PermissionResult.GRANTED -> requiresPermission.invoke()
                PermissionResult.DENIED -> onPermissionDenied?.invoke()
                PermissionResult.DENIED_AND_DISABLED -> onNeverAskAgain?.invoke()
                else -> Unit
            }
        })
    }
```

the **key** is [android.permission.READ_PHONE_STATE, android.permission.WRITE_EXTERNAL_STORAGE, android.permission.READ_EXTERNAL_STORAGE]

but **it[key]** get is null , becasuse Observer`s **it** the key is [android.permission.READ_EXTERNAL_STORAGE, android.permission.READ_PHONE_STATE, android.permission.WRITE_EXTERNAL_STORAGE] ,  **key** is different from **it** the key. 

so can not callback requiresPermission or onPermissionDenied or onNeverAskAgain  when  multiple permissions apply at the same time

**Comment by YassKnight** [comment]
I have the same problem

**Comment by hotchemi** [comment]
Do you mean there're some cases that the order of the permissions could be different? 👀

**Comment by hotchemi** [comment]
it should be resolved in 1.1.3, thx for your report!
